mypy/server/deps.py

```diff
@@ -235,6 +233,13 @@ def process_type_info(self, info: TypeInfo) -> None:
             self.add_type_dependencies(info.typeddict_type, target=make_trigger(target))
         if info.declared_metaclass:
             self.add_type_dependencies(info.declared_metaclass, target=make_trigger(target))
+        if info.is_protocol:
+            for base_info in info.mro[:-1]:
+                self.add_dependency(make_wildcard_trigger(base_info.fullname()),
```

Comment:
6 tests fail if I only add this for the class but not MRO. The idea is that we can have subprotocols, so that an actual interface is determined by the whole MRO, if something is changed in an explicit superprotocol we need to invalidate the protocol. A simple example is `[InvalidateProtocolViaSuperClass]`. There are other ways how to treat this (like including protocol members in a class snapshot), but it seems to me this one is conceptually simplest and the most similar to how no-protocol classes are treated.